department-of-veterans-affairs/vets-api
# [Search] Apply page limit based on search.gov offset limit

## Description of change


This PR fixes a minor bug on va.gov/search. Due to a limitation with the `offset` property on search.gov searches, the max page a user can visit is page 99. This is accounted for. However, if the user manually changes the page query param to 100, the current page goes haywire. 

### Example issue

https://www.va.gov/search/?page=100&query=test

![image](https://user-images.githubusercontent.com/1915775/70092128-95074180-15eb-11ea-8de9-2f621abefcf5.png)

Underlying JSON -
![image](https://user-images.githubusercontent.com/1915775/70092167-a94b3e80-15eb-11ea-8bbf-c8ef25bdfc48.png)

### How this fixes it
I really don't follow this pagination logic well, but from what I understand, search.gov returns a value `next_offset` that we use to derive the `current_page`. However, `next_offset` is `nil` when it reaches the search.gov offset limitation, `999`. This messes up that logic for deriving the `current_page`. This PR fixes this issue by accounting for the `total_viewable_pages` property as an additional max value for `current_page`.

There are still pagination issues here. For example, since 99 is the last viewable page, then results 981-990 are the last viewable search results, but the API's `totalEntries` property says 999. I tinkered with fixing this but it seems a little complicated. I think page 100 (result 991-1000) should be accessible though.

Ticket - https://github.com/department-of-veterans-affairs/va.gov-team/issues/2268

## Testing done

On localhost using a search.gov access key w/the FE running -
- Navigate to http://localhost:3001/search/?page=100&query=test
- Observe that the search results range doesn't go haywire

## Testing planned
This is pretty minor, so not much more besides testing in Staging before the deployment.

## Acceptance Criteria (Definition of Done)

#### Unique to this PR

- [x] I needed a search.gov access key to be able to test this locally

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)
